include/deal.II/base/polynomials_vector_anisotropic.h

```diff
@@ -0,0 +1,158 @@
+// ------------------------------------------------------------------------
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2004 - 2023 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// Part of the source code is dual licensed under Apache-2.0 WITH
+// LLVM-exception OR LGPL-2.1-or-later. Detailed license information
+// governing the source code and code contributions can be found in
+// LICENSE.md and CONTRIBUTING.md at the top level directory of deal.II.
+//
+// ------------------------------------------------------------------------
+
+#ifndef dealii_polynomials_vector_anisotropic_h
+#define dealii_polynomials_vector_anisotropic_h
+
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/polynomial.h>
+#include <deal.II/base/polynomial_space.h>
+#include <deal.II/base/tensor.h>
+#include <deal.II/base/tensor_polynomials_base.h>
+#include <deal.II/base/tensor_product_polynomials.h>
+
+#include <mutex>
+#include <vector>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * This class implements vector-valued anisotropic polynomials and can be
+ * used further for the generation of vector-valued polynomials with
+ * degrees being different in tangential and normal directions. The known
+ * examples include Raviart-Thomas and Nédélec polynomials.
```

Comment:
Maybe you can add one more sentence (either here or in the constructor) that says something like `The polynomial space can be renumbered, and in fact Raviart-Thomas and Nédélec elements will choose a different ordering to ensure normal or tangential continuity, respectively.`